src/ZprimePreSelectionModule.cxx

```diff
@@ -65,12 +66,19 @@ ZprimePreSelectionModule::ZprimePreSelectionModule(Context & ctx){
   muo_cleaner.reset(new MuonCleaner(AndId<Muon>(MuonIDTight(), PtEtaCut(45., 2.1))));
   ele_cleaner.reset(new ElectronCleaner(AndId<Electron>(ElectronID_PHYS14_25ns_tight_noIso, PtEtaSCCut(50., 2.5))));
 
-  jet_corrector.reset(new JetCorrector(JERFiles::PHYS14_L123_MC));
-  jetlepton_cleaner.reset(new JetLeptonCleaner(JERFiles::PHYS14_L123_MC));
+  is_mc = ctx.get("dataset_type") == "MC";
+  if (is_mc) {
+    jet_corrector.reset(new JetCorrector(JERFiles::Summer15_50ns_L123_AK4PFchs_MC));
+    jetlepton_cleaner.reset(new JetLeptonCleaner(JERFiles::Summer15_50ns_L123_AK4PFchs_MC));
+    topjet_corrector.reset(new TopJetCorrector(JERFiles::Summer15_50ns_L123_AK8PFchs_MC));
+  } else {
+    jet_corrector.reset(new JetCorrector(JERFiles::Summer15_50ns_L123_AK4PFchs_DATA));
+    jetlepton_cleaner.reset(new JetLeptonCleaner(JERFiles::Summer15_50ns_L123_AK4PFchs_DATA));
+    topjet_corrector.reset(new TopJetCorrector(JERFiles::Summer15_50ns_L123_AK8PFchs_DATA));
+  }
   jetlepton_cleaner->set_drmax(.4);
   jet_cleaner.reset(new JetCleaner(30., 2.4));
 
-  topjet_corrector.reset(new TopJetCorrector(JERFiles::PHYS14_L123_AK8PFchs_MC));
```

Comment:
For AK8 jets there are also new corrections available, see  https://github.com/UHH2/UHH2/blob/master/common/include/JetCorrections.h#L15-L16.